src/main/scala/it/turingtest/spotify/scala/client/SearchApi.scala

```diff
@@ -0,0 +1,99 @@
+package it.turingtest.spotify.scala.client
+
+import javax.inject.Inject
+import scala.concurrent.Future
+
+import com.vitorsvieira.iso.ISOCountry.ISOCountry
+import it.turingtest.spotify.scala.client.entities._
+import it.turingtest.spotify.scala.client.logging.AccessLogging
+import play.api.libs.json.{JsError, JsString, JsSuccess, Json, Reads}
+
+class SearchApi @Inject()(api: BaseApi) extends AccessLogging {
+
+  /**
+   * @see https://developer.spotify.com/web-api/search-item/
+   */
+  private final val SEARCH = s"${api.BASE_URL}/search"
+
+  /**
+   * Get Spotify catalog information about artists, albums, tracks or
+   * playlists that match a keyword string.
+   * @param q The search query's keywords (and optional field filters and
+   *          operators), for example q=roadhouse%20blues.
+   * @param itemType A comma-separated list of item types to search across.
+   *                 Valid types are: album, artist, playlist, and track.
+   * @param market An ISO 3166-1 alpha-2 country code or the string from_token.
+   * @param limit The maximum number of results to return.
+   * @param offset The index of the first result to return.
+   */
+  def search(
+    q: String,
+    itemType: SearchApi.ItemType,
```

Comment:
The API seems to enable the search specifying a list of types too. I think it could be done in an other search method which will return a less precise type, something like:
```
case class MultipleSearchResult(
  albums: Option[Page[SimpleAlbum]],
  artists: Option[Page[SimpleArtist]],
  playlists: Option[Page[SimplePlaylist]],
  tracks: Option[Page[Track]]
)
```
or maybe a `Map`. But this type is much less friendly as when manipulating it, we do not know what we searched for...

Reply:
Hi first of all thanks for taking the time to improve this project! Yes between the two I think the Option way is better, from a user point of view would probably be best to have a more specific type back, say if I request a Search for "album,artist" getting back just those two Page objects, but I think it would be kind of messy to implement.. For now I'll merge this and push a new release!